src/LibChorus/VcsDrivers/Mercurial/HgResumeTransport.cs

```diff
@@ -710,14 +710,10 @@ public bool Pull(string[] baseRevisions)
 				}
 				if (response.Status == PullStatus.Fail)
 				{
-					// Not sure that we need to abort the attempts just because .Net web request says so.  See Push also. CP 2012-06
-					errorMessage = "Pull data chunk failed";
-					_progress.WriteError(errorMessage);
```

Comment:
Seems like WriteError has a whole heap of power if that can cause the pull to fail. :-/  Looks more like a WriteErrorAndIndicateTotalFailureOfWholeProcess to me.  Would be better if those were to be separated at some point I think.